cli/src/lib.rs

```diff
@@ -1852,6 +1870,12 @@ fn idl(cfg_override: &ConfigOverride, subcmd: IdlCommand) -> Result<()> {
     }
 }
 
+fn get_idl_account(client: &RpcClient, idl_address: &Pubkey) -> Result<IdlAccount> {
+    let account = client.get_account(idl_address)?;
+    let mut data: &[u8] = &account.data;
+    return AccountDeserialize::try_deserialize(&mut data).map_err(|e| anyhow!("{:?}", e));
```

Comment:
CI failing due to this line https://github.com/coral-xyz/anchor/actions/runs/5013003528/jobs/8986288730?pr=2486#step:8:444

```suggestion
    AccountDeserialize::try_deserialize(&mut data).map_err(|e| anyhow!("{:?}", e))
```

Reply:
+1. I forgot to run the clippy on this. Interestingly my localhost run fails then on

```
cargo clippy --all-targets -- -D warnings
    Checking anchor-cli v0.27.0 (/home/chalda/my-testing/anchor/cli)
error: use of `unwrap_or` followed by a function call
    --> cli/src/lib.rs:1535:10
     |
1535 |         .unwrap_or(PathBuf::from("."))
     |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|| PathBuf::from("."))`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
     = note: `-D clippy::or-fun-call` implied by `-D warnings`

error: could not compile `anchor-cli` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `anchor-cli` due to previous error
```

that was not edited in this PR and was not reported in CI. Could be caused by different rust version?